### ISSUE TITLE
Maintainer's note: I am short on time to maintain this library. Contributions welcome!

### ISSUE THREAD

**Issue by jlevers** [jlevers/selling-partner-api#710]
Hey all,

I know I've been neglecting this repository a bit – life has gotten hectic, and it's hard to find time for it. When I first released it I never anticipated it would get millions of downloads. The fact that it has totally blows my mind.

Part of my motivation for releasing v6 was to make it easier for others to contribute. Previous versions were technically open-source, but the generator code wasn't, which made it virtually impossible for anyone but me to work on it. I'll put together a CONTRIBUTING.md doc sometime soon with an overview of how the generator code works.

If there are bugs that you want fixed and I'm not getting to em, I welcome your PRs! It might take me a little while to review them, but I _will_ review them eventually. Or, if you'd like to pay me to fix a bug or make an improvement, I'm open to that too – shoot me an email at [firstname] [at] highsidelabs [dot] co.

Thanks for using this library – I hope you've found it helpful :)

Jesse

**Comment by misterakko** [comment]
If I may chime in…
All of you probably know by now that Amazon has deprecated every type of feeds except for JSON ones. They plan on allowing JSON only from March 2025 onwards. Since MWS times, I have only used flat files — because that's what you get from a SQL database if you SELECT INTO FILE, basically.
Amazon does have a 
I was wondering if somebody here would be interested in writing with me an addendum on top the jlevers library which basically abstracts over the Amazon APIs and allows to easily send new product listings, and updates for price and stock for existing ones. Ideally, when the new code is robust, we could make a merge request here. The jdlevers library already allows one to make short work of authentication and document uploads, adding product handling would be par for the course IMHO.
I tried following Amazon's tutorial here https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide but it seems straitjacketed to building a user interface which, like Seller Central itself, forces the end user to type in one product at a time, field by field. If you have companies feeding you a catalog of tens of thousand of items, like I do, this way of doing things is simply unfeasible.

But maybe there's already some useful sample code, or code snippet, around and I just have not seen it? Let's talk.

**Comment by jlevers** [comment]
Hey @misterakko – I've actually started building something that would automatically convert the old feed formats to the new JSON format, so you wouldn't have to worry about updating how you're dealing with products, although I was planning on releasing it as a paid product. That said, if you get something out the door before I do and release it for free, it's all good! I'm all for more open source software :)

I've also already written some code for my own use for handling new product creation, so if you're interested in that, feel free to shoot me an email: [firstname] [at] highsidelabs [dot] co

**Comment by jlevers** [comment]
Just to follow up on this for anyone else who is trying to figure out how to convert from the old feed formats to the new JSON_LISTINGS_FEED format: I've built a product that handles this for you, here: https://tools.highsidelabs.co

You buy one-time access, install some Composer packages, and then input your old feed and get out the new JSON feed contents as an array. Full documentation here: https://docs.highsidelabs.co/feed-transformer

Also, just noting that for the `POST_PRODUCT_DATA` and `POST_FLAT_FILE_LISTINGS_DATA` feeds, it's looking like I'm going to have to provide the conversion service as an API rather than a downloadable package. There's just a bit too much infrastructure required for the product-type-specific nature of those feeds to work as a one-time download. More on this to come.

**Comment by HenrijsS** [comment]
Since Amazon released their own SDK for PHP: https://github.com/amzn/selling-partner-api-sdk/tree/main/php
Do you plan to continue this package?

**Comment by jlevers** [comment]
Yep, still planning to maintain this one. Theirs is far from feature-complete, and is using the default Swagger/OpenAPI codegen setup, which is what this library used before v6...I think it's a pretty limited tool for building good PHP SDKs. I talked with the team that's building it, and they said they have no real intention of their library superseding this one (we'll see how true that is), and I feel that this library has gathered enough steam at this point to be the natural default library for people to use.
